### PR TITLE
feat(geo): support Language options in map searches

### PR DESCRIPTION
**Pull request by roelvandenbrand**
#### Description of changes

Added a optional language attribute to the Geo search mapper to be able to receive the response from AWS Location services in a desired language.

#### Issue #, if available
https://github.com/aws-amplify/amplify-js/issues/11177

#### Description of how you validated changes
Added the optional language attribute to an existing test and verified it is in the request to the Location provider on AWS.
Ran "yarn test" 
Checked failed tests, there were some on Cognito where nothing changed. No error in the Geo tests.

#### Checklist
- [x] PR description included
- [x] `yarn test` passes (for Geo component)
- [x] Tests are changed (https://github.com/aws-amplify/amplify-js/blob/main/CONTRIBUTING.md#steps-towards-contributions)
- _Relevant documentation is changed or added (and PR referenced)_

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by roelvandenbrand** [comment]
Not sure which documentation to change. Much of it looks generated. And because of the line-numbers in the referencing docs some of it might need to be regenerate.

**Review by AllanZhengYP — APPROVED** [review]
The change looks good to me. Confirmed the new parameter is supported by underlying AWS SDK client. We just need to review the public interface change on Amplify library side.

File: .gitignore
[the remark this answers is not in the document]
Comment (roelvandenbrand) [reply .gitignore:1]: I think I reverted to the previous version of that file.

**Comment by roelvandenbrand** [comment]
I keep updating the branch, but is there something I should still do on the PR?

**Review by cwomack — APPROVED** [review]

**Review by thaddmt — APPROVED** [review]
LGTM!

**Review by cwomack — APPROVED** [review]

**Review by AllanZhengYP — APPROVED** [review]
Looks good to me!

**Comment by cwomack** [comment]
@roelvandenbrand, thank you for submitting this PR! We've gotten it merged and appreciate your patience and contributions.

**Pull request merged**